src/main/java/com/fasterxml/jackson/core/StreamReadConstraints.java

```diff
@@ -22,16 +27,37 @@
      */
     public static final int DEFAULT_MAX_STRING_LEN = 1_000_000;
 
+    protected final int _maxDepth;
```

Comment:
I suspect we need separate settings for reading and writing; this is for reading.

Reply:
This is StreamReadConstraints. We'll eventually need a StreamWriteConstraints but I think that should be done later.

Reply:
Ah yes. I realized that last night, right you are.